### ISSUE TITLE
Adding/updating app settings on existing App Service throws exception

### ISSUE THREAD

**Issue by shahiddev** [pulumi/examples#498]
**Pulumi CLI version 1.6.1**
**nuget package "1.7.0-preview-alpha.1575889533"**

I ran into an issue when updating an app service to add some app settings using .NET. 
I get an error similar to #476 

I've been able to repro using the [azure-cs-appservice-test example](https://github.com/pulumi/examples/tree/master/azure-cs-appservice)

**Repro steps:**
1. Deploy the stack as per instructions - should deploy fine
2. Modify the app settings to add a new setting (it doesn't matter what you put in the setting).
e.g. 
```
                AppSettings =
                {
                    { "WEBSITE_RUN_FROM_PACKAGE", codeBlobUrl },
                     { "test", codeBlobUrl }
                },
```
3. run Pulumi preview and you get an exception:
```
pulumi:pulumi:Stack (azure-cs-appservice-test):
    error: Running program 'c:\Repos\pulumi\examples\azure-cs-appservice\bin\Debug\netcoreapp3.0\Azure.AppService.dll' failed with an unhandled exception:
    System.InvalidOperationException: Expected System.String but got System.Collections.Immutable.ImmutableArray`1[[System.Object, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]] deserializing Pulumi.Azure.AppService.AppService.appSettings
       at Pulumi.Serialization.Converter.EnsureType[T](String context, Object val)
       at Pulumi.Serialization.Converter.ConvertObject(String context, Object val, Type targetType)
       at Pulumi.Serialization.Converter.ConvertDictionary(String fieldName, Object val, Type targetType)
       at Pulumi.Serialization.Converter.ConvertObject(String context, Object val, Type targetType)
       at Pulumi.Serialization.Converter.ConvertValue(String context, Value value, Type targetType)
       at Pulumi.Deployment.CompleteResourceAsync(Resource resource, ResourceArgs args, ResourceOptions options, ImmutableDictionary`2 completionSources)
       at Pulumi.Output`1.Pulumi.IOutput.GetDataAsync()
       at Pulumi.Output`1.Pulumi.IOutput.GetDataAsync()
       at Pulumi.Serialization.Serializer.SerializeAsync(String ctx, Object prop)
       at Pulumi.Serialization.Serializer.SerializeAsync(String ctx, Object prop)
       at Pulumi.Deployment.SerializeFilteredPropertiesAsync(String label, Input`1 args, Predicate`1 acceptKey)
       at Pulumi.Deployment.SerializeAllPropertiesAsync(String label, Input`1 args)
       at Pulumi.Deployment.RegisterResourceOutputsAsync(Resource resource, Output`1 outputs)
       at Pulumi.Deployment.Runner.WhileRunningAsync()
```

Same error occurs if the app service is created without any app settings and then you try and update to add some app settings.

The key step is that you have to be modifying an existing resource (provisioned by pulumi)

**Comment by mikhailshilkov** [comment]
@shahiddev Could you try updating `Pulumi.Azure` NuGet package to `1.7.0-preview`, if you aren't there yet?

**Comment by shahiddev** [comment]
@mikhailshilkov I see the same issue using the 1.7.0-preview package.

**Comment by mikhailshilkov** [comment]
This is solved in today's 1.8.0 Azure provider release
`<PackageReference Include="Pulumi.Azure" Version="1.8.0-preview" />`

It took us a while to reconcile some breaking changes, sorry for the mess.
